src/Core/src/Platform/Android/MauiHybridWebViewClient.cs

```diff
@@ -147,5 +148,116 @@ internal void Disconnect()
 		{
 			_handler.SetTarget(null);
 		}
+
+		private class DotNetInvokeAsyncStream : Stream
+		{
+			private const int PauseThreshold = 32 * 1024;
+			private const int ResumeThreshold = 16 * 1024;
+
+			private readonly Pipe _pipe;
+			private readonly Task<byte[]?> _task;
+
+			private bool _isDisposed;
+
+			public override bool CanRead => !_isDisposed;
+
+			public override bool CanSeek => false;
+
+			public override bool CanWrite => !_isDisposed;
+
+			public override long Length => throw new NotSupportedException();
+
+			public override long Position
+			{
+				get => throw new NotSupportedException();
+				set => throw new NotSupportedException();
+			}
+
+			public DotNetInvokeAsyncStream(Task<byte[]?> invokeTask)
+			{
+				_task = invokeTask;
+
+				_pipe = new Pipe(new PipeOptions(
+					pauseWriterThreshold: PauseThreshold,
+					resumeWriterThreshold: ResumeThreshold,
+					useSynchronizationContext: false));
+
+				StartReading();
+			}
+
+			private async void StartReading()
```

Comment:
This async void concerns me if the stream throws. I'm imagining a scenario where the method being invoked throws an exception from a property getter and then this crashes the whole process or something?